### PR TITLE
clickhouse-cityhash, clickhouse-driver, clickhouse-sqlalchemy

### PR DESCRIPTION
**Pull request by kszucs**
Initially generated with conda skeleton pypi.

ref mymarilyn/clickhouse-driver#6

### CODE REVIEW

**Comment by kszucs** [comment]
@conda-forge/staged-recipes please review

File: recipes/clickhouse-cityhash/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+about:
+  home: https://github.com/xzkostyan/python-cityhash
+  license: MIT
+  summary: 'Python-bindings for CityHash, a fast non-cryptographic hash algorithm'
+  license_family: MIT
+
+extra:
+  recipe-maintainers:
+    - kszucs
+    - xzkostyan
```
Comment (synapticarbors) [inline recipes/clickhouse-cityhash/meta.yaml]: @xzkostyan -- are you ok being added as a co-maintainer of the various clickhouse recipes on conda-forge listed in this PR?

File: recipes/clickhouse-cityhash/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package:
```
Comment (synapticarbors) [inline recipes/clickhouse-cityhash/meta.yaml]: Please consider using jinja templating variables as in:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L3-L19

File: recipes/clickhouse-cityhash/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  name: clickhouse-cityhash
+  version: "1.0.2"
+
+source:
+  fn: clickhouse-cityhash-1.0.2.tar.gz
+  url: https://pypi.python.org/packages/fb/e3/282f812aeace32817a6b2b894ccdfbe31146364294668cc85e1ad87b6f3f/clickhouse-cityhash-1.0.2.tar.gz
```
Comment (synapticarbors) [inline recipes/clickhouse-cityhash/meta.yaml]: Consider using the canonical url shown in:
https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L18

File: recipes/clickhouse-cityhash/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  md5: 13982de613981509c68377d328238304
```
Comment (synapticarbors) [inline recipes/clickhouse-cityhash/meta.yaml]: Please use the `sha256` hash

File: recipes/clickhouse-cityhash/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+requirements:
+  build:
+    - python
+    - setuptools
+
+  run:
+    - python
+
+test:
+  # Python imports
```
Comment (synapticarbors) [inline recipes/clickhouse-cityhash/meta.yaml]: remove comments

File: recipes/clickhouse-cityhash/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  url: https://pypi.python.org/packages/fb/e3/282f812aeace32817a6b2b894ccdfbe31146364294668cc85e1ad87b6f3f/clickhouse-cityhash-1.0.2.tar.gz
+  md5: 13982de613981509c68377d328238304
+
+build:
+  number: 0
+  skip: True  # [win]
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
```
Comment (synapticarbors) [inline recipes/clickhouse-cityhash/meta.yaml]: It looks like this builds C-extensions so please add `toolchain` as a `build` requirement.

File: recipes/clickhouse-cityhash/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    - python
+    - setuptools
+
+  run:
+    - python
+
+test:
+  # Python imports
+  imports:
```
Comment (synapticarbors) [inline recipes/clickhouse-cityhash/meta.yaml]: It seems like you should be importing `cityhash` here, but I'm not 100% sure.

File: recipes/clickhouse-cityhash/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    - clickhouse_driver
+    - clickhouse_driver.util
+
+about:
+  home: https://github.com/xzkostyan/python-cityhash
+  license: MIT
```
Comment (synapticarbors) [inline recipes/clickhouse-cityhash/meta.yaml]: Please add `license_file: LICENSE`

File: recipes/clickhouse-driver/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - six
```
Comment (synapticarbors) [inline recipes/clickhouse-driver/meta.yaml]: I don't think `six` is needed to build the package.

File: recipes/clickhouse-sqlalchemy/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  fn: clickhouse-sqlalchemy-0.0.3.tar.gz
+  url: https://pypi.python.org/packages/cb/3c/3720527c06c31ca1c2c1f0e33af30b3dc099e6a69d73357063a5f19126df/clickhouse-sqlalchemy-0.0.3.tar.gz
+  md5: b3e9eef9cd4960d1d7de47c843e7e987
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
```
Comment (synapticarbors) [inline recipes/clickhouse-sqlalchemy/meta.yaml]: I believe that only `setuptools` and `python` are `build` requirements of this package. The `--single-version-externally-managed` flag means that the `install_requires` aren't necessary unless `setup.py` causes them to be imported.

File: recipes/clickhouse-cityhash/meta.yaml (hunk 1 above)
Comment (xzkostyan) [reply recipes/clickhouse-cityhash/meta.yaml]: Yes, there is no problem.

File: recipes/clickhouse-cityhash/meta.yaml (hunk 2 above)
Comment (kszucs) [reply recipes/clickhouse-cityhash/meta.yaml]: @synapticarbors like this?

File: recipes/clickhouse-cityhash/meta.yaml (hunk 6 above)
Comment (kszucs) [reply recipes/clickhouse-cityhash/meta.yaml]: Added.

File: recipes/clickhouse-sqlalchemy/meta.yaml (hunk 10 above)
Comment (kszucs) [reply recipes/clickhouse-sqlalchemy/meta.yaml]: Should I remove `--single-version-externally-managed` from each of the packages?

Comment (synapticarbors) [reply recipes/clickhouse-sqlalchemy/meta.yaml]: No, you want that flag if `setup.py` uses `setuptools`. What I was trying to say is that the flag means that you don't need to list those other packages as `build` dependencies.  They should still appear as `run` dependencies.

**Comment by kszucs** [comment]
Thank You!

**Pull request merged**